src/inc/til/u8u16convert.h

```diff
@@ -269,62 +56,119 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
     // - S_OK          - the conversion succeeded
     // - E_OUTOFMEMORY - the function failed to allocate memory for the resulting string
     // - E_ABORT       - the resulting string length would exceed the upper boundary of an int and thus, the conversion was aborted before the conversion has been completed
-    // - E_UNEXPECTED  - an unexpected error occurred
-    template<class inT, class outT>
-    [[nodiscard]] typename std::enable_if<std::is_same<typename inT::value_type, char>::value && std::is_same<typename outT::value_type, wchar_t>::value, HRESULT>::type
-    u8u16(const inT in, outT& out) noexcept
+    // - E_UNEXPECTED  - the underlying conversion function failed
+    // - HRESULT value converted from a caught exception
+    template<class outT>
+    [[nodiscard]] HRESULT u8u16(const std::string_view& in, outT& out) noexcept
     {
         try
         {
             out.clear();
-
-            if (in.empty())
-            {
-                return S_OK;
-            }
+            RETURN_HR_IF(S_OK, in.empty());

```

Comment:
Audit mode is OK with this? I thought wil does or used to get mad when you returned success codes with the macros.... maybe its just not this one.

Reply:
It doesn't complain locally.